hrm-domain/hrm-core/case/caseRoutesV0.ts

```diff
@@ -93,16 +92,10 @@ casesRouter.put('/:id/status', canEditCase, async (req, res) => {
     body: { status },
   } = req;
   const { id } = req.params;
-  const updatedCase = await caseApi.updateCaseStatus(
-    id,
-    status,
-    accountSid,
-    user.workerSid,
-    {
-      can: req.can,
-      user,
-    },
-  );
+  const updatedCase = await caseApi.updateCaseStatus(id, status, accountSid, {
+    can: req.can,
+    user,
+  });
   if (!updatedCase) {
```

Comment:
I'm not keen on that idea. As far as the service layer is concerned, not finding something isn't an error, it just returns nothing because it found nothing which is pretty standard in APIs.

HTTP treats not finding things as an error, but that's HTTPs concern, other consumers might not, forcing the service API to throw not found errors still feels like pushing HTTP specific concerns onto the service layer.

More broadly the Result type should only be used when explicit error handling is needed, if it's fine for the default top level error handling to take care of things, or error states can be inferred in higher levels of the API from reasonable non error return values, we should do that.